### ISSUE TITLE
Figure out why these two commits cause a failure without a clean

### ISSUE THREAD

**Issue by jgkamat** [RoboJackets/robocup-software#899]
Right now,
`git checkout e40271418f000f1cce7d423e1af2ae935bc2e31a && make clean all && git checkout master && make`
fails. 

This is probably an issue with RRT's or robocup-software's cmake lists, and results in these errors:
```
[25/29] Building CXX object soccer/CMakeFiles/robocup.dir/planning/RRTUtil.cpp.o
FAILED: ccache /usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -Drobocup_EXPORTS -Isrc/googletest/include -I../common -Icommon -isystem /usr/include/eigen3 -Isoccer/rrt/include -I../soccer -Isoccer -isystem /usr/include/libusb-1.0 -isystem /usr/include/SDL2 -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtOpenGL -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtSvg -isystem /usr/include/python3.4m -Werror=return-local-addr -fPIC   -fPIC -std=gnu++14 -MD -MT soccer/CMakeFiles/robocup.dir/planning/RRTUtil.cpp.o -MF soccer/CMakeFiles/robocup.dir/planning/RRTUtil.cpp.o.d -o soccer/CMakeFiles/robocup.dir/planning/RRTUtil.cpp.o -c /home/jay/Code/robocup-software/soccer/planning/RRTUtil.cpp
/home/jay/Code/robocup-software/soccer/planning/RRTUtil.cpp: In function ‘void Planning::DrawRRT(const RRT::Tree<Geometry2d::Point>&, SystemState*, unsigned int)’:
/home/jay/Code/robocup-software/soccer/planning/RRTUtil.cpp:36:18: error: request for member ‘parent’ in ‘node’, which is of pointer type ‘RRT::Node<Geometry2d::Point>* const’ (maybe you meant to use ‘->’ ?)
         if (node.parent()) {
                  ^
/home/jay/Code/robocup-software/soccer/planning/RRTUtil.cpp:37:42: error: request for member ‘state’ in ‘node’, which is of pointer type ‘RRT::Node<Geometry2d::Point>* const’ (maybe you meant to use ‘->’ ?)
             state->drawLine(Segment(node.state(), node.parent()->state()),
                                          ^
/home/jay/Code/robocup-software/soccer/planning/RRTUtil.cpp:37:56: error: request for member ‘parent’ in ‘node’, which is of pointer type ‘RRT::Node<Geometry2d::Point>* const’ (maybe you meant to use ‘->’ ?)
             state->drawLine(Segment(node.state(), node.parent()->state()),
                                                        ^
[25/29] Building CXX object soccer/CMakeFiles/robocup.dir/planning/RRTPlanner.cpp.o
FAILED: ccache /usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -Drobocup_EXPORTS -Isrc/googletest/include -I../common -Icommon -isystem /usr/include/eigen3 -Isoccer/rrt/include -I../soccer -Isoccer -isystem /usr/include/libusb-1.0 -isystem /usr/include/SDL2 -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtOpenGL -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtSvg -isystem /usr/include/python3.4m -Werror=return-local-addr -fPIC   -fPIC -std=gnu++14 -MD -MT soccer/CMakeFiles/robocup.dir/planning/RRTPlanner.cpp.o -MF soccer/CMakeFiles/robocup.dir/planning/RRTPlanner.cpp.o.d -o soccer/CMakeFiles/robocup.dir/planning/RRTPlanner.cpp.o -c /home/jay/Code/robocup-software/soccer/planning/RRTPlanner.cpp
/home/jay/Code/robocup-software/soccer/planning/RRTPlanner.cpp: In member function ‘std::vector<Geometry2d::Point> Planning::RRTPlanner::runRRT(Planning::MotionInstant, Planning::MotionInstant, const MotionConstraints&, const Geometry2d::ShapeSet&, SystemState*, unsigned int)’:
/home/jay/Code/robocup-software/soccer/planning/RRTPlanner.cpp:193:11: error: ‘class RRT::BiRRT<Geometry2d::Point>’ has no member named ‘setMinIterations’
     biRRT.setMinIterations(_minIterations);
           ^
ninja: build stopped: subcommand failed.
make: *** [makefile:18: all] Error 1
```

Running `make clean && make` resolves the issue, but I'd rather figure out why this happens.

**Comment by jgkamat** [comment]
See this file: https://github.com/RoboJackets/robocup-software/blob/8926d4d98e858aa023b82506de4870c877995780/soccer/CMakeLists.txt#L9

It seems like the externalproject_add is not performing the 'download' step every time, and this is causing the files to fail to update in the build step. This is my attempt (which dosen't work) at fixing it:

```
diff --git a/soccer/CMakeLists.txt b/soccer/CMakeLists.txt
index ce6faf2..f9229a0 100644
--- a/soccer/CMakeLists.txt
+++ b/soccer/CMakeLists.txt
@@ -10,15 +10,24 @@ set(rrt_install_dir ${CMAKE_CURRENT_BINARY_DIR}/rrt)
 set(rrt_lib_path ${rrt_install_dir}/lib/libRRT.a)
 ExternalProject_Add(rrt_project
     URL ${PROJECT_SOURCE_DIR}/external/rrt
+    SOURCE_DIR ${PROJECT_SOURCE_DIR}/external/rrt
     CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=${rrt_install_dir}
+    UPDATE_COMMAND ""
     BUILD_BYPRODUCTS ${rrt_lib_path}
+    BUILD_ALWAYS 1
+    BUILD_IN_SOURCE 1
 )
+ExternalProject_Add_Step(rrt_project
+    COMMAND ${CMAKE_COMMAND} -E echo "Force configure of myProject"
+    DEPENDEES update
+    DEPENDERS configure
+    ALWAYS 1)
 set_target_properties(rrt_project PROPERTIES EXCLUDE_FROM_ALL TRUE)
 
 # add rrt library target.  IMPORTED means that it is created externally
-add_library(rrt STATIC IMPORTED)
-set_property(TARGET rrt PROPERTY IMPORTED_LOCATION ${rrt_lib_path})
-add_dependencies(rrt rrt_project)
+# add_library(rrt STATIC IMPORTED)
+# set_property(TARGET rrt PROPERTY IMPORTED_LOCATION ${rrt_lib_path})
+# add_dependencies(rrt rrt_project)
 
 # include rrt library headers
 include_directories(${rrt_install_dir}/include/)
@@ -146,7 +155,8 @@ target_link_libraries(robocup GL GLU glut)
 target_link_libraries(robocup pthread)
 target_link_libraries(robocup spnav)
 target_link_libraries(robocup ${SDL2_LIBRARIES})
-target_link_libraries(robocup rrt)
+target_link_libraries(robocup ${rrt_lib_path})
+add_dependencies(robocup rrt_project)
 
 # python
 # note: these are set in the root CMakeLists.txt file

```

**Comment by jgkamat** [comment]
The best solution is probably to add the rrt repo as a subdirectory, which is the 'proper' way to do it. The flags seem to clobber between the two repos, so that needs to be sorted out.

**Comment by jgkamat** [comment]
https://github.com/RoboJackets/robocup-software/tree/jay/fixbuild

See this branch for progress

**Comment by jgkamat** [comment]
Should be closed by #914. Let me know if you still see this issue.
